### PR TITLE
Docs update - Add protocol specific API reference pages. 

### PR DESCRIPTION
**Pull request by soos3d**
Added:
* `ethereum-api-reference.md`.
* `arbitrum-api-reference.md`.
* `polygon-api-reference.md`.
* `avalanche-api-reference.md`.

Fixed:
* `node-api-reference.md`.
  * Fixed inconsistencies and quotes. 

Added new pages to `config.js`

### CODE REVIEW

File: docs/api/arbitrum-api-reference.md [hunk 1]
```diff
@@ -0,0 +1,2180 @@
...
+
+Arbitrum is a Layer 2 scaling solution for the Ethereum network that leverages optimistic rollups to introduce several efficiency improvements to transaction processing and their costs. To do this, Arbitrum bundles several off-chain transactions together in a batch before parsing the result to the Ethereum network. 
+
+## Bridge assets between Ethereum and Arbitrum
+
+You can bridge between Ethereum and Arbitrum using the [Arbitrum brigde](https://bridge.arbitrum.io/?l2ChainId=42161). The process is straightforward, you can connect MetaMask, and it takes around ten minutes to complete.
+
+### Use the Arbitrum testnet
+
+Chainstack supports the Arbitrum testnet on Goerli. You can deploy an Arbitrum Goerli testnet node, then bridge some Goerli ETH using the [testnet Arbitrum brigde](https://bridge.arbitrum.io/?l2ChainId=421613).
```
Comment (akegaviar) [inline docs/api/arbitrum-api-reference.md:32]: Link to our Arbitrum tools docs section.

File: docs/api/arbitrum-api-reference.md [hunk 2]
```diff
@@ -0,0 +1,2180 @@
...
+This page is a collection of JSON-RPC API call examples for the Arbitrum API using:
+
+* [web3.js](https://web3js.readthedocs.io/).
+* [web3.py](https://web3py.readthedocs.io/).
+* [eth.rb](https://github.com/q9f/eth.rb/). 
+* [cURL](https://curl.se/).
+
+## What is Arbitrum
+
+[Arbitrum](https://developer.offchainlabs.com/) is an EVM-compatible sidechain, and one of Ethereum's [scaling solutions](https://chainstack.com/solving-the-blockchain-trilemma-scaling-solutions-for-ethereum/) to help reduce costs and increase efficiency.
```
Comment (akegaviar) [inline docs/api/arbitrum-api-reference.md:22]: Not a sidechain, it's a rollup.

File: docs/api/arbitrum-api-reference.md (hunk 1 above)
Comment (akegaviar) [inline docs/api/arbitrum-api-reference.md:32]: `Arbitrum Goerli testnet`

File: docs/api/arbitrum-api-reference.md [hunk 3]
```diff
@@ -0,0 +1,2180 @@
...
+
+### Use the Arbitrum testnet
+
+Chainstack supports the Arbitrum testnet on Goerli. You can deploy an Arbitrum Goerli testnet node, then bridge some Goerli ETH using the [testnet Arbitrum brigde](https://bridge.arbitrum.io/?l2ChainId=421613). 
+ 
+## What is the Arbitrum API
+
+The Arbitrum API allows developers to communicate with the Arbitrum blockchain to build applications. 
+
+To read data from and send transactions to the Arbitrum blockchain, an application must connect to an Arbitrum node.
```
Comment (akegaviar) [inline docs/api/arbitrum-api-reference.md:38]: @py-goh do you want `Arbitrum node` to be an `Arbitrum RPC node`?

**Comment by akegaviar** [comment]
We also need to remove `Node API reference` I think now that we targeting to have a separate page for each protocol.

**Comment by akegaviar** [comment]
Protocol order:

* Ethereum API
* Polygon API
* Avalanche API
* Arbitrum API

**Comment by akegaviar** [comment]
The see also sections are not rendering

SCR-20220908-bq9

File: docs/api/avalanche-api-reference.md [hunk 4]
```diff
@@ -0,0 +1,2192 @@
...
+This page is a collection of JSON-RPC API call examples for the Avalanche API using:
+
+* [web3.js](https://web3js.readthedocs.io/).
+* [web3.py](https://web3py.readthedocs.io/).
+* [eth.rb](https://github.com/q9f/eth.rb/). 
+* [cURL](https://curl.se/).
+
+## What is Avalanche
+
+[Avalanche](https://www.avax.network/) is an EVM-compatible protocol focused on speed and scalability. Avalanche is the fastest smart contracts platform in the blockchain industry, measured by time-to-finality.
```
Comment (akegaviar) [inline docs/api/avalanche-api-reference.md:22]: Remove this:

```
Avalanche is the fastest smart contracts platform in the blockchain industry, measured by time-to-finality.
```
to keep it neutral

File: docs/api/avalanche-api-reference.md [hunk 5]
```diff
@@ -0,0 +1,2192 @@
...
+
+The Avalanche network can achieve high throughput, estimated to reach over 4500 transactions per second. Transactions are finalized in under a second while being able to scale to unprecedented levels. The protocol allows for high transactions per second (TPS) throughput while having low latency and consuming extremely little energy, switching to a low-energy-consumption state when there is no activity.
```
Comment (akegaviar) [inline docs/api/avalanche-api-reference.md:24]: Remove this section too.

File: docs/api/polygon-api-reference.md [hunk 6]
```diff
@@ -0,0 +1,2155 @@
...
+This page is a collection of JSON-RPC API call examples for the Polygon API using:
+
+* [web3.js](https://web3js.readthedocs.io/).
+* [web3.py](https://web3py.readthedocs.io/).
+* [eth.rb](https://github.com/q9f/eth.rb/). 
+* [cURL](https://curl.se/).
+
+## What is Polygon POS
+
+[Polygon POS](https://docs.polygon.technology/docs/develop/getting-started) is an EVM-compatible sidechain, and one of Ethereum's [scaling solutions](https://chainstack.com/solving-the-blockchain-trilemma-scaling-solutions-for-ethereum/) to help reduce costs and increase efficiency.
```
Comment (akegaviar) [inline docs/api/polygon-api-reference.md:22]: We are renaming `Polygon PoS` to `Polygon` everywhere for simplicity, so drop `POS`

File: docs/api/polygon-api-reference.md [hunk 7]
```diff
@@ -0,0 +1,2155 @@
...
+
+Polygon POS runs independently from the Ethereum chain, has its own consensus mechanism, and can bridge assets between the two networks. 
+
+This allows Ethereum developers to leverage lower transactions cost and faster transaction speed to scale applications while maintaining the security offered by the Ethereum network.
```
Comment (akegaviar) [inline docs/api/polygon-api-reference.md:26]:
```
while maintaining the security offered by the Ethereum network.
```
I don't think so. It's a separate chain. So just remove the paragraph.

**Review by akegaviar — CHANGES_REQUESTED** [review]
Changes please

File: docs/api/arbitrum-api-reference.md (hunk 3 above)
Comment (py-goh) [reply docs/api/arbitrum-api-reference.md:38]: Arbitrum RPC node would be great

Comment (akegaviar) [reply docs/api/arbitrum-api-reference.md:38]: @soos3d for all four then please

Comment (py-goh) [reply docs/api/arbitrum-api-reference.md:38]: All [protocol] nodes are preferred to be [protocol] RPC nodes

File: docs/api/arbitrum-api-reference.md (hunk 1 above)
Comment (soos3d) [reply docs/api/arbitrum-api-reference.md:32]: How did I not think about that

File: docs/api/arbitrum/arbitrum-api-reference.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+[Eth.rb](https://github.com/q9f/eth.rb/) is a Ruby gem that allows you to communicate with an EVM node through HTTP.
+
+To install, run:
+
+```sh
+gem install eth
+```
+
+::: tip See also
```
Comment (akegaviar) [inline docs/api/arbitrum/arbitrum-api-reference.md:92]: No full stop, a bullet point, and blank lines please:

```
::: tip See also

* <a href="https://chainstack.com/build-better-with-arbitrum/" target="_blank">Build better with Arbitrum</a>

:::
```

File: docs/api/avalanche/avalanche-api-reference.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+[Eth.rb](https://github.com/q9f/eth.rb/) is a Ruby gem that allows you to communicate with an EVM node through HTTP.
+
+To install, run:
+
+```sh
+gem install eth
+```
+
+::: tip See also
```
Comment (akegaviar) [inline docs/api/avalanche/avalanche-api-reference.md:90]: same

File: docs/api/polygon/polygon-api-reference.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+[Eth.rb](https://github.com/q9f/eth.rb/) is a Ruby gem that allows you to communicate with an EVM node through HTTP.
+
+To install, run:
+
+```sh
+gem install eth
+```
+
+::: tip See also
```
Comment (akegaviar) [inline docs/api/polygon/polygon-api-reference.md:84]: same

**Review by akegaviar — CHANGES_REQUESTED** [review]
Minor changes

**Comment by akegaviar** [comment]
@py-goh have a look please

**Comment by akegaviar** [comment]
Also @soos3d resolve conflicts please

**Comment by py-goh** [comment]
- Separated URL path LGTM
- Backlinks to BBW Protocol LGTM
- Protocol API Reference still using hash but it's fine since the doc is short. If the Protocol API Reference happens to expand into a super long doc, separated path is preferred

**Review by akegaviar — CHANGES_REQUESTED** [review]
One more change please. Hopefully final.

Since we'll be at The Merge when this PR is merged, add to all Ethereum API methods that take `latest`, `earliest`, `pending` as parameters the two post merge ones: `safe` and `finalized`. Now they accept all 5: latest, earliest, pending, safe, finalized.

**Comment by akegaviar** [comment]
> One more change please. Hopefully final.

> Since we'll be at The Merge when this PR is merged, add to all Ethereum API methods that take latest, earliest, pending as parameters the two post merge ones: safe and finalized. Now they accept all 5: latest, earliest, pending, safe, finalized.

Actually, let's make it as a separate PR.

**Review by akegaviar — APPROVED** [review]
LGTM

**Review by easeev — CHANGES_REQUESTED** [review]
Rebase / fix tests

**Review by akegaviar — APPROVED** [review]
lgtm

**Pull request merged**